src/Microsoft.NetCore.Analyzers/Core/Security/DoNotOverloadDeserializationBinderWithoutThrowingAnException.cs

```diff
@@ -0,0 +1,158 @@
+                    WellKnownTypeProvider typeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+                    INamedTypeSymbol SerializationBinder = null;
+                    if (!typeProvider.TryGetTypeByMetadataName("System.Runtime.Serialization.SerializationBinder", out SerializationBinder))
+                    {
+                        System.Diagnostics.Debug.Fail("Unable to find System.Runtime.Serialization.SerializationBinder threw WellKnownTypeProvider");
+                        return;
+                    }
+
+                    compilationStartAnalysisContext.RegisterSymbolAction(
+                        (SymbolAnalysisContext symbolAnalysisContext) =>
+                        {
+                            IMethodSymbol msym = (IMethodSymbol)symbolAnalysisContext.Symbol;
+                            if (msym.Name == "BindToType")
+                            {
+                                if (ITypeSymbolExtensions.Inherits(msym.ContainingType, SerializationBinder))
+                                {
+                                    IBlockOperation msymOps = msym.GetTopmostOperationBlock(compilationStartAnalysisContext.Compilation);
+                                    methodCache[msym] = CacheState.InProgress;
+                                    if (!hasThrow(msymOps.Operations))
+                                    {
+                                        symbolAnalysisContext.ReportDiagnostic(
+                                            Diagnostic.Create(
+                                                DoNotOverloadSerializationBinderWithoutThrowingAnExceptionRule,
+                                                msym.Locations[0],
+                                                msym.ContainingType.ToDisplayString(
+                                                    SymbolDisplayFormat.MinimallyQualifiedFormat)));
+                                    }
+                                }
+                            }
+                        }
+                    , SymbolKind.Method);
+
+                    bool hasThrow(IEnumerable<Microsoft.CodeAnalysis.IOperation> ops)
+                    {
+                        if (ops == null)
+                        {
+                            //doesn't happen, but just in case
+                            System.Diagnostics.Debug.Fail("hasThrow called on null value. It should be called on an IEnumerable of IOperations.");
+                            return false;
+                        }
+
+                        foreach (IOperation op in ops)
+                        {
+                            if (op == null)
+                            {
+                                continue;
+                            }
+                            if (op.Kind == OperationKind.Invocation)
+                            {
```

Comment:
Nit: Add blank lines to separate code blocks that aren't logically related.

All the if statements below as well.